opm/autodiff/FlowMain.hpp

```diff
@@ -391,6 +391,8 @@ namespace Opm
                 deck_ = parser->parseFile(deck_filename, parseContext);
                 checkDeck(deck_, parser);
                 eclipse_state_.reset(new EclipseState(deck_, parseContext));
+                auto ioConfig = eclipse_state_->getIOConfig();
+                ioConfig->setOutputDir(output_dir_);
```

Comment:
No reasonable cases, no, with the exception of sub objects which use lazy instantiation. These already handle this though and behave correctly under constness.